py/mpprint.h

```diff
@@ -79,4 +79,44 @@ int mp_printf(const mp_print_t *print, const char *fmt, ...);
 int mp_vprintf(const mp_print_t *print, const char *fmt, va_list args);
 #endif
 
+// Debug messages during code developing.
+// An approximate hierarchy of debug levels is:
+// -1 - SUPPRES all messages in release version
+// 0 - the most CRITICAL errors, often requiring a system reset, use message with this level if possible raising an exception
+// 1 - ERROR requiring program restart, use message with this level before raising an exception
+// 2 - WARNING, something went wrong, but you can fix it with additional operations in code right now
+// 3 - INFO, it is interesting and useful for understanding a bug
+// 4 - DEBUG, more detaled info, dig deeper
+// 5 - TRACE, show algorithm flow
+// In real you may use own classification of debug levels.
+#define DBG(level, ...) \
```

Comment:
Should be guarded by `#ifndef DBG` because this is a super common name and people might want to change what it does.

Reply:
It may be PRN() & PRN_LEVEL

Reply:
That's still fairly common. Could prefix it with `mp_`. Still should be a no-op by default I guess, as are all other macros like this (DEBUG_printf etc)

Reply:
Both pair (PRN() & PRN_LEVEL) and (MSG() & MSG_LEVEL) not used in MicroPython.

There is 
```
#define debug(lvl, x...) do { if (lvl <= DEBUG_LEVEL) { printf(x); } } while (0)
```
at https://github.com/atgreen/libffi/blob/e9de7e35f2339598b16cbb375f9992643ed81209/src/pa/ffi.c#L52

Reply:
> Both pair (PRN() & PRN_LEVEL) and (MSG() & MSG_LEVEL) not used in MicroPython.

Indeed, but mpprint.h is part of the public API so any definitions it in might clash with external software. So if someone includes mpprint.h, in that same file it would not be possible to also include another file which has MSG defined (unless working around it)

Reply:
Changed to PM_PRN() & PM_PRN_LEVEL